eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/PaymentOnion.scala

```diff
@@ -264,9 +267,19 @@ object PaymentOnion {
 
       object Blinded {
         def validate(records: TlvStream[OnionPaymentPayloadTlv], blindedRecords: TlvStream[RouteBlindingEncryptedDataTlv], nextBlinding: PublicKey): Either[InvalidTlvPayload, Blinded] = {
+          if (records.get[EncryptedRecipientData].isEmpty) return Left(MissingRequiredTlv(UInt64(10)))
+          // Bolt 4: MUST return an error if the payload contains other tlv fields than `encrypted_recipient_data` and `current_blinding_point`.
           if (records.get[AmountToForward].nonEmpty) return Left(ForbiddenTlv(UInt64(2)))
           if (records.get[OutgoingCltv].nonEmpty) return Left(ForbiddenTlv(UInt64(4)))
```

Comment:
These two lines are now redundant with the more generic check for all TLVs.

Reply:
That's true, the only advantage they have is that they provide an error message that correctly mentions the tlv tag (2 or 4), whereas the general check will just provide 0, but that's probably not worth keeping, I'll remove it.

Reply:
Removed in https://github.com/ACINQ/eclair/pull/2408/commits/f83d5de31bebe5aa0a1b195d9856ecb98ba2a9e4